src/Enclosure.php

```diff
@@ -251,12 +249,8 @@ public function __construct($link = null, $type = null, $length = null, $javascr
         $this->type = $type;
         $this->width = $width;
 
-        if (class_exists('idna_convert')) {
-            $idn = new \idna_convert();
-            $parsed = \SimplePie\Misc::parse_url($link);
-            $this->link = \SimplePie\Misc::compress_parse_url($parsed['scheme'], $idn->encode($parsed['authority']), $parsed['path'], $parsed['query'], $parsed['fragment']);
```

Comment:
The point of using `idna_convert` is that the user can type a domain name containing Unicode characters (so called [IDN](https://en.wikipedia.org/wiki/Internationalized_domain_name)) and it is encoded as ASCII (using [Punycode](https://en.wikipedia.org/wiki/Punycode) scheme), which various system APIs actually expect.

Though it looks like cURL, at least, can do this conversion too:

```php
php > $ch = curl_init('https://háčkyčárky.cz'); curl_setopt($ch, CURLOPT_RETURNTRANSFER, true); curl_exec($ch); var_dump(curl_error($ch));
string(0) ""
```

Unlike native PHP functions:

```php
php > var_dump(file_get_contents('https://háčkyčárky.cz'));

Warning: file_get_contents(): php_network_getaddresses: getaddrinfo for háčkyčárky.cz failed: Name or service not known in php shell code on line 1

Warning: file_get_contents(https://háčkyčárky.cz): Failed to open stream: php_network_getaddresses: getaddrinfo for háčkyčárky.cz failed: Name or service not known in php shell code on line 1
bool(false)
```

Reply:
Oh, it seems that I mixed that up. 😅 Thank you, I will correct that.

Reply:
I fixed this by using the correct classes and methods to encode/convert the Unicode domain to ASCII.